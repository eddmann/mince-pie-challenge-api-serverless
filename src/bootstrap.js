// @flow

import bootstrap from './handlers/bootstrap';

const { USER_POOL_ID, USER_POOL_CLIENT_ID, BASE_ENDPOINT_URL } = process.env;

if (!USER_POOL_ID) {
  throw new Error('USER_POOL_ID is not present');
}

if (!USER_POOL_CLIENT_ID) {
  throw new Error('USER_POOL_CLIENT_ID is not present');
}

if (!BASE_ENDPOINT_URL) {
  throw new Error('BASE_ENDPOINT_URL is not present');
}

export const handler = bootstrap({
  getPoolId: () => USER_POOL_ID,
  getClientId: () => USER_POOL_CLIENT_ID,
  getBaseEndpointUrl: () => BASE_ENDPOINT_URL,
});
