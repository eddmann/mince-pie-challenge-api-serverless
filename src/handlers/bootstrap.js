/* @flow */

import { ok } from '../helpers/http';
import { createHandler } from '../helpers/handlers';
import { Resource } from 'hal';

const bootstrap = async ({ callback, services: { env } }) => {
  const resource = new Resource(
    {
      cognito: {
        poolId: env.USER_POOL_NAME.split('/')[1],
        clientId: env.USER_POOL_CLIENT_ID,
      },
      baseEndpointUrl: env.BASE_ENDPOINT_URL,
    },
    '/',
  );

  resource.link('list', '/pies');
  resource.link('add', '/pies');
  resource.link('view', { href: '/pies/{id}', templated: true });

  callback(undefined, ok(resource));
};

export default createHandler(bootstrap);
