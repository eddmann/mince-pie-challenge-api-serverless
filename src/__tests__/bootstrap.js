import bootstrap from '../handlers/bootstrap';

const response = () =>
  function cb(error, response) {
    if (error) fail(`Unexpected error: ${error}`);
    cb.response = response;
  };

it('displays bootstrap details', async () => {
  const services = {
    env: {
      USER_POOL_NAME: 'eu-west-1_POOL_NAME',
      USER_POOL_CLIENT_ID: 'CLIENT_ID',
      BASE_ENDPOINT_URL: 'https://base.url',
    },
  };
  const callback = response();

  await bootstrap(services)({}, {}, callback, undefined);

  expect(callback.response).toMatchSnapshot();
});
