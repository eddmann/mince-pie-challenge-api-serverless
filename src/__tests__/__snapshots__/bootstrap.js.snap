// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays bootstrap details 1`] = `
Object {
  "body": "{\\"_links\\":{\\"self\\":{\\"href\\":\\"/\\"},\\"list\\":{\\"href\\":\\"/pies\\"},\\"add\\":{\\"href\\":\\"/pies\\"},\\"view\\":{\\"href\\":\\"/pies/{id}\\",\\"templated\\":true}},\\"cognito\\":{\\"clientId\\":\\"CLIENT_ID\\"},\\"baseEndpointUrl\\":\\"https://base.url\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": "*",
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/hal+json",
  },
  "statusCode": 200,
}
`;
